src/btree/bt_sync.c

```diff
@@ -499,11 +499,12 @@ __wt_sync_file(WT_SESSION_IMPL *session, WT_CACHE_OP syncop)
             }
 
             /*
-             * Write dirty pages, if we can't skip them. If we skip a page, mark the tree dirty. The
-             * checkpoint marked it clean and we can't skip future checkpoints until this page is
-             * written.
+             * Write dirty pages, if we can't skip them except for history store. As part of
+             * checkpoint, history store gets populated, so we shouldn't ignore the changes in this
+             * checkpoint. If we skip a page, mark the tree dirty. The checkpoint marked it clean
+             * and we can't skip future checkpoints until this page is written.
              */
-            if (__sync_checkpoint_can_skip(session, page)) {
+            if (!is_hs && __sync_checkpoint_can_skip(session, page)) {
```

Comment:
I'd move this to `__sync_checkpoint_can_skip`, it already has a list of conditions for which it returns false, and this could be added to that list rather than testing it separately.

I don't understand why the dirty hs pages are different from other files, I would have expected the test against `first_dirty_txn` to catch this case, but is it that we're not writing the hs store with the correct transaction IDs? Anyway, maybe the comment could explain that further?

Reply:
I am fine to move the check inside the function.

It used to catch the scenarios with `first_dirty_txn` until the history store inserts are done without a transaction. As part of the checkpointing the data store, we will move the history values into the history store. The `first_dirty_txn` of the page is set with `txn_global.last_running`.

In a scenario where no operations are performed during the checkpoint even, we refresh the snapshot before flushing the history store, all the modifications in the history store are more than the `snap_max`. So the history store updates are never flushed to disk. This is the main reason behind the skip of history store updates.

@keithbostic Do you see any problem in this approach?

Reply:
That makes sense to me, and I'd welcome a more expansive comment if you can.

Reply:
Fixed by adding a check in the `__sync_checkpoint_can_skip` function. Added a comment to explain reason behind not to skip history store changes.